pkgs/by-name/bo/bolt-launcher/package.nix

```diff
@@ -0,0 +1,177 @@
+          stdenv.cc.cc.lib
+        ];
+        rpath = lib.makeLibraryPath [
+          glib
+          nss
+          nspr
+          atk
+          at-spi2-atk
+          libdrm
+          expat
+          xorg.libxcb
+          libxkbcommon
+          xorg.libX11
+          xorg.libXcomposite
+          xorg.libXdamage
+          xorg.libXext
+          xorg.libXfixes
+          xorg.libXrandr
+          mesa
+          gtk3
+          pango
+          cairo
+          alsa-lib
+          dbus
+          at-spi2-core
+          cups
+          xorg.libxshmfence
+          systemd
+        ];
+      in
+      ''
+        mkdir -p $out/lib/ $out/share/cef/
+        cp libcef_dll_wrapper/libcef_dll_wrapper.a $out/lib/
+        cp -r ../Resources/* $out/lib/
+        cp -r ../Release/* $out/lib/
+        patchelf --set-rpath "${rpath}" $out/lib/libcef.so
+        patchelf --set-rpath "${gl_rpath}" $out/lib/libEGL.so
+        patchelf --set-rpath "${gl_rpath}" $out/lib/libGLESv2.so
+        cp ../Release/*.bin $out/share/cef/
+        cp -r ../Resources/* $out/share/cef/
+        cp -r ../include $out
+        cp -r ../libcef_dll $out
+        cp -r ../cmake $out
+      '';
+  });
+in
+let
+  bolt = stdenv.mkDerivation (finalAttrs: {
+    pname = "bolt-launcher";
+    version = "0.9.0";
+
+    src = fetchFromGitHub {
+      owner = "AdamCake";
+      repo = "bolt";
+      rev = finalAttrs.version;
+      fetchSubmodules = true;
+      hash = "sha256-LIlRDcUWbQwIhFjtqYF+oVpTOPZ7IT0vMgysEVyJ1k8=";
+    };
+
+    nativeBuildInputs = [
+      cmake
+      ninja
+      luajit
+      makeWrapper
+    ];
+
+    buildInputs = [
+      mesa
+      xorg.libX11
+      xorg.libxcb
+      libarchive
+      libz
+      cef
+      jdk17
+    ];
+
+    cmakeFlags = [
+      "-D CMAKE_BUILD_TYPE=Release"
+      "-D BOLT_LUAJIT_INCLUDE_DIR=${luajit}/include"
+      "-G Ninja"
+    ];
+
+    preConfigure = ''
+      mkdir -p cef/dist/Release cef/dist/Resources cef/dist/include
+
+      ln -s ${cef}/lib/* cef/dist/Release
+
+      ln -s ${cef}/share/cef/*.pak cef/dist/Resources
+      ln -s ${cef}/share/cef/icudtl.dat cef/dist/Resources
+      ln -s ${cef}/share/cef/locales cef/dist/Resources
+
+      ln -s ${cef}/include/* cef/dist/include
+      ln -s ${cef}/libcef_dll cef/dist/libcef_dll
+
+      ln -s ${cef}/cmake cef/dist/cmake
+      ln -s ${cef}/CMakeLists.txt cef/dist
+    '';
+
+    postFixup = ''
+      makeWrapper "$out/opt/bolt-launcher/bolt" "$out/bin/${finalAttrs.pname}-${finalAttrs.version}" \
+      --set JAVA_HOME "${jdk17}"
+      ls -al $out/bin
+      mkdir -p $out/lib
+      cp $out/usr/local/lib/libbolt-plugin.so $out/lib
+    '';
+  });
+in
+buildFHSEnv {
+  inherit (bolt) name version;
+
+  targetPkgs =
+    pkgs:
+    [ bolt ]
+    ++ (with pkgs; [
+      xorg.libSM
+      xorg.libXxf86vm
+      xorg.libX11
+      glib
+      pango
+      cairo
+      gdk-pixbuf
+      gtk2-x11
+      libz
+      libcap
+      libsecret
+      openssl_1_1
```

Comment:
Very unlikely.A lot of packages using `openssl_1_1` usually hard depend on it, which from my understanding and testing is also the case with `bolt-launcher`. I'd love to find a better solution but I think this might unfortunately be an issue with every package that depends on it.

Reply:
openssl 1.1 and libcrypto 1.1 are import dependencies of one of the games Bolt can launch. I made it an optdepend on AUR since not everyone wants to play that game. I did also notify Jagex that they need to update this and got an acknowledgement, but... well, don't hold your breath.

Reply:
> Could we put it behind a feature flag, so that the default will not be marked as insecure immediately?

I'm not sure if that would be reasonable. As @Adamcake mentioned, those insecure packages are unfortunately needed so that RS3 can run in the first place. I don't think providing a feature flag that breaks half the app when set to false would help a lot, leaving it as insecure would be better.

Besides, the proper RS3 package in nixpkgs is also broken because of it, as is cinny and a whole lot of other apps, so this should be probably okay.

Reply:
I imagine there are many people who want to install Bolt and are fine with it not being able to launch RS3. I think the feature flag approach makes sense, it would allow people to avoid needing insecure dependencies to use the launcher for other clients.

Obviously not up to me to make any decision on this -- but I took a stab at this here to use on my local if you are interested in pursuing this approach feel free to merge it in or to ignore if uninterested!

https://github.com/nezia1/nixpkgs/pull/1

Reply:
Thanks a lot @tomking ! We should be good to go now. We ended up adding an `enableRS3` flag that is set to false by default, as well as adding a desktop entry and an icon.